common/src/edgeai_camera.cpp

```diff
@@ -0,0 +1,144 @@
+
+    tiovx_sensor_module_params_init(&sensorObj);
+    sensorObj.is_interactive = 0;
+    sensorObj.ch_mask = chMask;
+
+    tiovx_sensor_module_query(&sensorObj);
+
+    /* App works only for IMX390 2MP cameras */
+    if(camInputInfo->m_sen_id == "imx390")
+    {
+        sensorObj.sensor_index = 0;
+    }
+    else
+    {
+        throw runtime_error("App works only for IMX390 2MP cameras now. \n");
+    }
+
+    tiovx_capture_module_params_init(&captureObj, &sensorObj);
+    captureObj.enable_error_detection = 0; // or 1
+    captureObj.out_bufq_depth = 4; // This must be greater than 3
+
+    /* VISS Module params init */
+    tivx_vpac_viss_params_init(&vissObj.params);
+
+    snprintf(vissObj.dcc_config_file_path,
+             TIVX_FILEIO_FILE_PATH_LENGTH,
+             viss_dcc_path.c_str());
+
+    vissObj.input.bufq_depth = 1;
+
+    vissObj.input.params.width              = sensorObj.image_width;
+    vissObj.input.params.height             = sensorObj.image_height;
+    vissObj.input.params.num_exposures      = sensorObj.sensorParams.sensorInfo.raw_params.num_exposures;
+    vissObj.input.params.line_interleaved   = sensorObj.sensorParams.sensorInfo.raw_params.line_interleaved;
+    vissObj.input.params.meta_height_before = sensorObj.sensorParams.sensorInfo.raw_params.meta_height_before;
+    vissObj.input.params.meta_height_after  = sensorObj.sensorParams.sensorInfo.raw_params.meta_height_after;
+    vissObj.input.params.format[0].pixel_container = sensorObj.sensorParams.sensorInfo.raw_params.format[0].pixel_container;
+    vissObj.input.params.format[0].msb      = sensorObj.sensorParams.sensorInfo.raw_params.format[0].msb;
+    vissObj.ae_awb_result_bufq_depth        = 1;
+
+    vissObj.output_select[0] = TIOVX_VISS_MODULE_OUTPUT_NA; // 0
+    vissObj.output_select[1] = TIOVX_VISS_MODULE_OUTPUT_NA;
+    vissObj.output_select[2] = TIOVX_VISS_MODULE_OUTPUT_EN; // 1
+    vissObj.output_select[3] = TIOVX_VISS_MODULE_OUTPUT_NA;
+    vissObj.output_select[4] = TIOVX_VISS_MODULE_OUTPUT_NA;
+
+    /* As we are selecting output2, specify output2 image properties */
+    vissObj.output2.bufq_depth   = 1;
+    vissObj.output2.color_format = VX_DF_IMAGE_NV12;
+    vissObj.output2.width        = sensorObj.image_width;
+    vissObj.output2.height       = sensorObj.image_height;
+
+    vissObj.h3a_stats_bufq_depth = 1;
+
+    /* LDC Module params init */
+    snprintf(ldcObj.dcc_config_file_path,
+             TIVX_FILEIO_FILE_PATH_LENGTH,
+             ldc_dcc_path.c_str());
+
+    /* Configuring LDC in DCC mode */
+    ldcObj.ldc_mode = TIOVX_MODULE_LDC_OP_MODE_DCC_DATA;
+    ldcObj.en_out_image_write = 0;
+    ldcObj.en_output1 = 0;
+
+    ldcObj.input.bufq_depth = 1;
+    ldcObj.input.color_format = VX_DF_IMAGE_NV12;
+    ldcObj.input.width = vissObj.output2.width;
+    ldcObj.input.height = vissObj.output2.height;
+
+    ldcObj.init_x = 0;
+    ldcObj.init_y = 0;
+    ldcObj.table_width = 1920;
```

Comment:
These should be parameterized from the camera chosen.

Reply:
Done. Updated the PR